src/core/core.scale.js

```diff
@@ -256,48 +256,47 @@ class Scale extends Element {
 		};
 		// TODO: make maxWidth, maxHeight private
 		/** @type {number} */
-		this.maxWidth = undefined;
+		this.maxWidth = -1;
 		/** @type {number} */
-		this.maxHeight = undefined;
+		this.maxHeight = -1;
 		/** @type {number} */
-		this.paddingTop = undefined;
+		this.paddingTop = -1;
 		/** @type {number} */
-		this.paddingBottom = undefined;
+		this.paddingBottom = -1;
 		/** @type {number} */
-		this.paddingLeft = undefined;
+		this.paddingLeft = -1;
 		/** @type {number} */
-		this.paddingRight = undefined;
+		this.paddingRight = -1;
 
 		// scale-specific properties
 		/** @type {string} */
 		this.axis = undefined;
-		/** @type {number} */
+		/** @type {number|undefined} */
 		this.labelRotation = undefined;
 		this.min = undefined;
 		this.max = undefined;
 		/** @type {object[]} */
-		this.ticks = null;
-		/** @type {object[]} */
+		this.ticks = [];
+		/** @type {object[]|null} */
```

Comment:
I think you could use:
```js
/** @type {object[]=} */
```
to indicate nullable object array

Reply:
Apparently that's the same as `object[]|undefined`